### ISSUE TITLE
discoverServices() function produces incorrect results

### ISSUE THREAD

**Issue by jameskentTX** [pazznetwork/ngx-chat#25]
The problem is because app doesn't process info nodes: `https://xmpp.org/extensions/xep-0030.html#info-nodes`

```typescript
private async discoverServices() {
        const serviceListResponsePromise = this.chatAdapter.chatConnectionService.sendIq(
            new QueryStanzaBuilder(
                ServiceDiscoveryPlugin.DISCO_ITEMS, this.chatAdapter.chatConnectionService.userJid.domain).toStanza()
        );

        const serviceDomains = (await serviceListResponsePromise)
            .getChild('query').getChildren('item').map(itemNode => itemNode.attrs.jid);

        const discoveredServices = await Promise.all(serviceDomains.map(serviceDomain => this.discoverServiceInformation(serviceDomain)));
        this.services.push(...discoveredServices);
 }
```

Code above maps according to `jid`, but info nodes in my case (or maybe always, I don't know) have the same `jid`.

My response from server:
```xml
<iq xmlns="jabber:client" xml:lang="en" to="admin@pc17.home/30959486515595480201154" from="pc17.home" type="result" id="info1">
            <query xmlns="http://jabber.org/protocol/disco#items">
                <item jid="conference.pc17.home"/>
                <item jid="proxy.pc17.home"/>
                <item jid="pubsub.pc17.home"/>
                <item jid="upload.pc17.home"/>
                <item node="announce" name="Announcements" jid="pc17.home"/>
                <item node="config" name="Configuration" jid="pc17.home"/>
                <item node="user" name="User Management" jid="pc17.home"/>
                <item node="online users" name="Online Users" jid="pc17.home"/>
                <item node="all users" name="All Users" jid="pc17.home"/>
                <item node="outgoing s2s" name="Outgoing s2s Connections" jid="pc17.home"/>
                <item node="running nodes" name="Running Nodes" jid="pc17.home"/>
                <item node="stopped nodes" name="Stopped Nodes" jid="pc17.home"/>
            </query>
        </iq>
```

```typescript
const serviceDomains = (await serviceListResponsePromise)
            .getChild('query').getChildren('item').map(itemNode => itemNode.attrs.jid);
```

`serviceDomains` result in my case:
`Array(12)
0: "conference.pc17.home"
1: "proxy.pc17.home"
2: "pubsub.pc17.home"
3: "upload.pc17.home"
4: "pc17.home"
5: "pc17.home"
6: "pc17.home"
7: "pc17.home"
8: "pc17.home"
9: "pc17.home"
10: "pc17.home"
11: "pc17.home"`

Then there is:
```typescript
const discoveredServices = await Promise.all(serviceDomains.map(serviceDomain => this.discoverServiceInformation(serviceDomain)));
```
which checks many times the same value `"pc17.home"`

Because program is working in such a way function `discoverServerFeatures()` produces `this.services` variable, which contain the same values in array. But `findService` function rejects results if length > 1:

```typescript
    findService(category: string, type: string): Promise<Service> {
        return new Promise((resolve, reject) => {
            this.servicesInitialized$.pipe(first(value => !!value)).subscribe(() => {
                const results = this.services.filter(service =>
                    service.identities.filter(identity => identity.category === category && identity.type === type).length > 0
                );
                if (results.length === 0) {
                    reject(`no service matching category ${category} and type ${type} found!`);
                } else if (results.length > 1) {
                    reject(`multiple services matching category ${category} and type ${type} found! ${JSON.stringify(results)}`);
                } else {
                    return resolve(results[0]);
                }
            });
        });
    }
```

My specific problem: if  `category === "pubsub"` and `type === "pep"` `findService` returns incorrect answer.

**Comment by trampi** [comment]
Hi @jameskentTX,
I will filter the duplicate requests by only sending one request per jid in `discoverServices`.
This should probably help with your problem.
Just to be sure, could you post the result for the `disco#info` calls? One response should suffice.

By the way, you should consider upgrading to Angular 9 if you haven't already. I would like to drop support for Angular 8 in the next release. This fix should however be easily applicable to the old version of ngx-chat if this is inconvenient :-)

**Comment by jameskentTX** [comment]
Thanks, Trampi.

```xml
<iq xmlns="jabber:client" xml:lang="en" to="admin@pc17.home/173283692429512504971410" from="pc17.home" type="result" id="info1">
    <query xmlns="http://jabber.org/protocol/disco#info">
        <identity type="pep" category="pubsub"/>
        <identity name="ejabberd" type="im" category="server"/>
        <feature var="http://jabber.org/protocol/commands"/>
        <feature var="http://jabber.org/protocol/disco#info"/>
        <feature var="http://jabber.org/protocol/disco#items"/>
        <feature var="http://jabber.org/protocol/offline"/>
        <feature var="http://jabber.org/protocol/pubsub"/>
        <feature var="http://jabber.org/protocol/pubsub#access-authorize"/>
        <feature var="http://jabber.org/protocol/pubsub#access-open"/>
        <feature var="http://jabber.org/protocol/pubsub#access-presence"/>
        <feature var="http://jabber.org/protocol/pubsub#access-whitelist"/>
        <feature var="http://jabber.org/protocol/pubsub#auto-create"/>
        <feature var="http://jabber.org/protocol/pubsub#auto-subscribe"/>
        <feature var="http://jabber.org/protocol/pubsub#collections"/>
        <feature var="http://jabber.org/protocol/pubsub#config-node"/>
        <feature var="http://jabber.org/protocol/pubsub#create-and-configure"/>
        <feature var="http://jabber.org/protocol/pubsub#create-nodes"/>
        <feature var="http://jabber.org/protocol/pubsub#delete-items"/>
        <feature var="http://jabber.org/protocol/pubsub#delete-nodes"/>
        <feature var="http://jabber.org/protocol/pubsub#filtered-notifications"/>
        <feature var="http://jabber.org/protocol/pubsub#get-pending"/>
        <feature var="http://jabber.org/protocol/pubsub#instant-nodes"/>
        <feature var="http://jabber.org/protocol/pubsub#item-ids"/>
        <feature var="http://jabber.org/protocol/pubsub#last-published"/>
        <feature var="http://jabber.org/protocol/pubsub#manage-subscriptions"/>
        <feature var="http://jabber.org/protocol/pubsub#member-affiliation"/>
        <feature var="http://jabber.org/protocol/pubsub#modify-affiliations"/>
        <feature var="http://jabber.org/protocol/pubsub#multi-items"/>
        <feature var="http://jabber.org/protocol/pubsub#outcast-affiliation"/>
        <feature var="http://jabber.org/protocol/pubsub#persistent-items"/>
        <feature var="http://jabber.org/protocol/pubsub#presence-notifications"/>
        <feature var="http://jabber.org/protocol/pubsub#presence-subscribe"/>
        <feature var="http://jabber.org/protocol/pubsub#publish"/>
        <feature var="http://jabber.org/protocol/pubsub#publish-only-affiliation"/>
        <feature var="http://jabber.org/protocol/pubsub#publish-options"/>
        <feature var="http://jabber.org/protocol/pubsub#publisher-affiliation"/>
        <feature var="http://jabber.org/protocol/pubsub#purge-nodes"/>
        <feature var="http://jabber.org/protocol/pubsub#retract-items"/>
        <feature var="http://jabber.org/protocol/pubsub#retrieve-affiliations"/>
        <feature var="http://jabber.org/protocol/pubsub#retrieve-default"/>
        <feature var="http://jabber.org/protocol/pubsub#retrieve-items"/>
        <feature var="http://jabber.org/protocol/pubsub#retrieve-subscriptions"/>
        <feature var="http://jabber.org/protocol/pubsub#shim"/>
        <feature var="http://jabber.org/protocol/pubsub#subscribe"/>
        <feature var="http://jabber.org/protocol/pubsub#subscription-notifications"/>
        <feature var="http://jabber.org/protocol/rsm"/>
        <feature var="iq"/>
        <feature var="jabber:iq:last"/>
        <feature var="jabber:iq:privacy"/>
        <feature var="jabber:iq:register"/>
        <feature var="jabber:iq:version"/>
        <feature var="msgoffline"/>
        <feature var="presence"/>
        <feature var="urn:xmpp:blocking"/>
        <feature var="urn:xmpp:carbons:2"/>
        <feature var="urn:xmpp:mam:0"/>
        <feature var="urn:xmpp:mam:1"/>
        <feature var="urn:xmpp:mam:2"/>
        <feature var="urn:xmpp:mam:tmp"/>
        <feature var="urn:xmpp:ping"/>
        <feature var="vcard-temp"/>
        <x type="result" xmlns="jabber:x:data">
            <field var="FORM_TYPE" type="hidden">
                <value>
                    http://jabber.org/network/serverinfo
                </value>
            </field>
        </x>
    </query>
</iq>
```

**Comment by trampi** [comment]
Should be fixed with 0.9.0.
